### PR TITLE
Update CI and publishing workflows

### PR DESCRIPTION
**Pull request by keturiosakys**
This small PR updates the CI and publishing workflows to allow publishing the plugin and library packages independently. 

- rename check wrappers
- split out publish to npm to separate actions triggered by (protected) tags

### CODE REVIEW

File: .github/workflows/publish_lib_to_npm.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Publish lib/ (and convenience package autometrics/) packages to NPMJS
+on:
+  push:
+    tags:
+      - "lib-*"
+jobs:
+  test:
```
Comment (hatchan) [inline .github/workflows/publish_lib_to_npm.yml:7]: Is this here as a requirement for the `publish` job? If so, then we should probably add the `needs: test` property to the `publish` job. (see: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)

File: .github/workflows/publish_typescript_plugin_to_npm.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Publish lib/ (and convenience package autometrics/) packages to NPMJS
```
Comment (hatchan) [inline .github/workflows/publish_typescript_plugin_to_npm.yml:1]: I think the name needs to be updated.

**Review by hatchan — APPROVED** [review]
Some minor things, other than that :+1:

**Review by hatchan — APPROVED** [review]
Some minor things, other than that :+1:

File: .github/workflows/publish_lib_to_npm.yml (hunk 1 above)
Comment (keturiosakys) [reply .github/workflows/publish_lib_to_npm.yml:7]: ah nice one! Updating now

**Pull request merged**